packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -482,6 +514,30 @@ export class TreeCheckout implements ITreeCheckoutFork {
 			this.events.emit("commitApplied", data, getRevertible);
 			withinEventContext = false;
 		});
+
+		// When the branch is trimmed, we can garbage collect any repair data whose latest relevant revision is one of the
+		// trimmed revisions.
+		branch.on("ancestryTrimmed", (revisions) => {
+			this.withCombinedVisitor((visitor) => {
+				revisions.forEach((revision) => {
+					this.removedRoots
+						// get all the roots last created or used by the revision
+						.getLatestRelevantRoots(revision)
+						// remove any roots not created by valid revertibles
+						.filter(
+							([_, { major }]) =>
+								major !== undefined && !this.revertibleCommitBranches.has(major),
+						)
+						// get the detached field for the root and delete it from the removed roots
+						.map(([root, detachedNodeId]) => {
+							const field = this.removedRoots.toFieldKey(root);
+							this.removedRoots.deleteEntry(detachedNodeId);
```

Comment:
When calling `map` on an array, it's best not to put code with side effects in the map function (because `map` is meant to be a functional programing idiom). You can instead put all of the code with side effects in the `forEach`.